here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/DeleteFeature.kt

```diff
@@ -0,0 +1,140 @@
+package naksha.psql.executors.write
+
+import naksha.model.*
+import naksha.model.Metadata.Metadata_C.hash
+import naksha.model.objects.NakshaFeature
+import naksha.model.request.ReadFeatures
+import naksha.model.request.SuccessResponse
+import naksha.model.request.Write
+import naksha.psql.PgCollection
+import naksha.psql.PgColumn
+import naksha.psql.PgSession
+import naksha.psql.PgTable
+import naksha.psql.PgUtil.PgUtilCompanion.quoteIdent
+import naksha.psql.executors.PgReader
+import naksha.psql.executors.write.WriteFeatureUtils.newFeatureTupleNumber
+import naksha.psql.executors.write.WriteFeatureUtils.resolveFlags
+import naksha.psql.executors.write.WriteFeatureUtils.tuple
+
+class DeleteFeature(session: PgSession) : UpdateFeature(session) {
+    override fun execute(collection: PgCollection, write: Write): Tuple {
+        val featureId = write.featureId
+        require(featureId != null) {"No feature ID provided"}
+
+        var feature = write.feature
+        if (feature == null) {
+            val readFeatures = ReadFeatures(collection.id)
+            readFeatures.featureIds.add(featureId)
+            val response = PgReader(session, readFeatures).execute().proxy(SuccessResponse::class)
+            if (!response.features.isEmpty())
+            {
+                feature = response.features.first()
+            }
+        }
+
+        val previousMetadata = fetchCurrentMeta(collection, featureId)
+        val tupleNumber = newFeatureTupleNumber(collection, featureId, session)
+        val flags = resolveFlags(collection, session).action(Action.DELETED)
+        val tuple = tuple(
+            session.storage,
+            tupleNumber,
+            feature,
+            metadata(tupleNumber, feature, featureId, flags),
+            write.attachment,
+            flags
+        )
```

Comment:
For what do we need the tuple? When deleting, we first copy from HEAD to HISTORY, then we copy again from HEAD to HISTORY and/or DELETED (the tombstone). We do not need to create any new tuples our self, because no data is modified, only minor metadata is changed. The tuple-number for the first copy stays unchanged. The tuple-number of the second copy will only get a new `txn` (version), `uid` (unique-id in version) and `flags`. We only need to return the tuple-number, which we can generate without any further knowledge. Nothing else (as well not the hash) do change!